### ISSUE TITLE
Loss not decreasing!

### ISSUE THREAD

**Issue by mahao01** [kmkurn/pytorch-crf#42]
I use CRF as my model loss like [Issue #29](https://github.com/kmkurn/pytorch-crf/issues/29), but I found loss didn't decrease! I replace it with `BCEWithLogitsLoss` and then loss decreases.
I have `y: (seq_length,)` and `y_pred: (seq_length, num_classes)`. Here is my code:
```
# features is a list here
for epoch in range(epochs):
    for fea in features:
        y_pred = model(fea)
        y_pred= y_pred.reshape(y_pred.shape[0], 1, -1)
        y= y.reshape(y.shape[0], -1)
        loss = -model_gat.crf(y_pred, y)
        optimizer.zero_grad()
        loss.backward()
        optimizer.step()
```
I wonder whether I use the library by mistake?
Looking for your help, thanks!

**Comment by kmkurn** [comment]
Hi, your code looks fine to me so I'm not sure why the loss didn't decrease. I've confirmed with a simple code that the CRF works fine. Could you post the smallest working code to reproduce the behaviour?

**Comment by mahao01** [comment]
Thanks very much, I'll check my code again.
